addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/internal/HS110.java

```diff
@@ -0,0 +1,147 @@
+        SWITCH_OFF("{\"system\":{\"set_relay_state\":{\"state\":0}}}}"),
+        SYSINFO("{\"system\":{\"get_sysinfo\":null}}"),
+        ENERGY("{\"emeter\":{\"get_realtime\":null}}");
+
+        public String value;
+
+        Command(String value) {
+            this.value = value;
+        }
+    }
+
+    public String ip;
+
+    public HS110(String ip) {
+        this.ip = ip;
+    }
+
+    public boolean isPresent() {
+
+        try {
+
+            InetAddress ip = InetAddress.getByName(this.ip);
+            return ip.isReachable(500);
+        } catch (IOException ex) {
+        }
+        return false;
+    }
+
+    public boolean sendSwitch(OnOffType command) throws IOException {
+        String jsonData = sendCommand(command == ON ? Command.SWITCH_ON : Command.SWITCH_OFF);
+        if (jsonData.length() > 0) {
+
+            JsonObject jo = new JsonParser().parse(jsonData).getAsJsonObject();
+            int errorCode = jo.get("system").getAsJsonObject().get("set_relay_state").getAsJsonObject().get("err_code")
+                    .getAsInt();
+            return errorCode == 0;
+        }
+        return false;
+    }
+
+    /* SysInfo relatet parsing */
+
+    public static JsonObject parseSysinfoObject(String data) {
+        JsonObject dataObject = new JsonParser().parse(data).getAsJsonObject();
+        JsonObject systemObject = dataObject.getAsJsonObject("system");
+        JsonObject sysInfo = systemObject.getAsJsonObject("get_sysinfo");
+        return sysInfo;
+    }
+
+    public static String parseSysinfo(String data) {
+        return parseSysinfoObject(data).toString();
+    }
+
+    public static String parseAddress(String data) {
+        return parseSysinfoObject(data).get("address").getAsString();
+    }
+
+    public static BigDecimal parsePort(String data) {
+        return parseSysinfoObject(data).get("port").getAsBigDecimal();
+
+    }
+
+    public static String parseDeviceId(String data) {
+        return parseSysinfoObject(data).get("deviceId").getAsString();
+
+    }
+
+    public static OnOffType parseState(String data) {
+        int state = parseSysinfoObject(data).get("relay_state").getAsInt();
+        return state == 1 ? OnOffType.ON : OnOffType.OFF;
+    }
+
+    /* Energy related parsing */
+    private static JsonObject parseEnergyObject(String data) {
+        return new JsonParser().parse(data).getAsJsonObject().getAsJsonObject("emeter").getAsJsonObject("get_realtime");
+    }
+
+    public static BigDecimal parseWattage(String data) {
+        return parseEnergyObject(data).get("power").getAsBigDecimal();
+    }
+
+    public static BigDecimal parseTotal(String data) {
+        return parseEnergyObject(data).get("total").getAsBigDecimal();
+    }
+
+    public String sendCommand(Command command) throws IOException, ConnectException {
+
+        logger.debug("Executing command {}", command.value);
+
+        Socket socket = new Socket(ip, HS100_PORT);
```

Comment:
Better use a [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) here because you might leave this socket open if an exception occurs